src/OpenTelemetry.Exporter.ApplicationInsights/Implementation/TraceExporterHandler.cs

```diff
@@ -341,13 +341,23 @@ private void OverwriteSpanKindFromAttribute(IAttributeValue spanKindAttr, ref Sp
             {
                 var kind = spanKindAttr.Match((s) => s, null, null, null, null);
 
-                if (kind == "server")
+                switch (kind.ToLower(CultureInfo.InvariantCulture))
                 {
-                    resultKind = SpanKind.Server;
-                }
-                else
-                {
-                    resultKind = SpanKind.Client;
+                    case "server":
+                        resultKind = SpanKind.Server;
+                        break;
+                    case "client":
+                        resultKind = SpanKind.Client;
+                        break;
+                    case "producer":
+                        resultKind = SpanKind.Producer;
```

Comment:
should we ever expect anyone to set attribute instead of Span kind? If there a real scenario for this? If no, I'd say we stop doing it going forward and make everyone set kind.

Reply:
appears that this would be related to support of tags: https://github.com/opentracing/specification/blob/master/semantic_conventions.md

Reply:
I'd suggest then that Opentracing shim would do it, otherwise, every exporter will need to support it forever going forward

Reply:
In that case much of the code in src\OpenTelemetry.Exporter.ApplicationInsights\Implementation\TraceExporterHandler.cs#ExportAsync would be relevant to move out to the OpenTracing shim.  Rather than remove code now I'd like to keep it in place for reference and open an issue to refactor this exporter when the shim work happens.

Reply:
Added comment to #28